parla/task_runtime.py

```diff
@@ -349,118 +347,163 @@ def result(self):
         elif isinstance(self._state, TaskException):
             raise self._state.exc
 
+    @abstractmethod
+    def _run(self) -> Optional[TaskState]:
+        """Execution of the task."""
+        raise NotImplementedError()
+    
+    @abstractmethod
+    def _post_run(self, ctx: 'SchedulerContext'):
+        """Cleanup works after executing the task."""
+        raise NotImplementedError()
```

Comment:
I like what you did here, but I think the naming is a bit confusing.

Can we keep the top-level `run` and rename the above to something like `_execute_work` and `_update_resources` perhaps? Or maybe something else. I just think calling `_run` from `run` is confusing.

Overall, great cleanup, reducing duplicating code and pulling methods into the base class.

Reply:
I agree with you, sean. I think  `cleanup_task_` would be better. Updating resources looks confusing since it could be happened both pre- and post- task execution.

Reply:
or `cleanup_resource()` something?

Reply:
Yeah and also `_run()` is confusing. as Sean suggested, `_execute_work` or `_execute_task`? (I prefer task since work and task sound similar and we are running a task)

Reply:
Thanks for the naming suggestions. I was not aware of the semantics and just put the ugly name.